modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -4910,9 +4851,10 @@ private static void parallelSortInIsolatedPool(
             }
         };
 
-        ForkJoinPool forkJoinPool = new ForkJoinPool(PARALLEL_SORT_THREADS + 1, factory, null, false);
+        ExecutorService execPool = pool == null ?
```

Comment:
Hmmmmm..........
Can you please put back that ForkJoinPool? Otherwise the default one will be used.
It was the whole point of introducing additional pool here - avoid using default one.